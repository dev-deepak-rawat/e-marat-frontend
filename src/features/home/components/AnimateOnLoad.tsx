import styled from 'styled-components';
import { useState, useEffect } from 'react';

export type PropsType = {
	animationStartClasses: string;
	children: React.ReactNode;
};

export default function AnimateOnLoad({
	animationStartClasses,
	children,
}: PropsType) {
	const [animationClasses, setAnimationClasses] = useState<string>(
		animationStartClasses
	);

	useEffect(() => {
		setAnimationClasses('');

		return () => {
			setAnimationClasses(animationStartClasses);
		};
	}, []);

	return <Root className={`transform ${animationClasses}`}>{children}</Root>;
}

const Root = styled.div`
	transition: transform 2s cubic-bezier(0, 1, 0.3, 1) 0.1s,
		opacity 0.7s ease-out 0.1s;
	will-change: transform, opacity;
`;
