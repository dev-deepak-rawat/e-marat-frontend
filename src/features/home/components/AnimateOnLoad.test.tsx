import AnimateOnLoad from 'features/home/components/AnimateOnLoad';
import renderer from 'react-test-renderer';

it('Animate On Load', () => {
	const tree = renderer
		.create(
			<AnimateOnLoad animationStartClasses="opacity-0">
				mock child
			</AnimateOnLoad>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
