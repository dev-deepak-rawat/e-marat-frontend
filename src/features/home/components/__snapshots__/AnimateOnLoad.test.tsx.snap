// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Animate On Load 1`] = `
<div
  className="sc-bdfBQB gTatkM transform opacity-0"
>
  mock child
</div>
`;
